src/main/scala/discord/model/DispatchEvent.scala

```diff
@@ -15,16 +17,78 @@ object DispatchEvent {
   implicit val config: Configuration = Configuration.default.withSnakeCaseMemberNames
 
   // TODO: Implement these types
-  type Activity     = Unit
-  type ChannelType  = Unit
-  type ClientStatus = Unit
-  type Emoji        = Unit
-  type GuildFeature = String
-  type GuildMember  = Unit
-  type Overwrite    = Unit
-  type Role         = Unit
-  type Snowflake    = Long
-  type VoiceState   = Unit
+  type ActivityFlags = Unit
+  type Assets        = Unit
+  type ChannelType   = Unit
+  type ClientStatus  = Unit
+  type GuildFeature  = String
+  type GuildMember   = Unit
+  type Overwrite     = Unit
+  type Party         = Unit
+  type Role          = Unit
+  type Secrets       = Unit
+  type Snowflake     = Long
+  type VoiceState    = Unit
+
+  // TODO: Move case classes that don't extend DispatchEvent out of here since this file is already gonna be huge
+  case class Activity(
+      name: String,
+      `type`: ActivityType,
+      url: Option[Uri],
+      createdAt: Timestamp,
+      timestamps: Timestamps,
+      applicationId: Snowflake,
+      details: Option[String],
+      state: Option[String],
+      emoji: Option[Emoji],
+      party: Party,
+      assets: Assets,
+      secrets: Secrets,
+      instance: Boolean,
+      flags: ActivityFlags
+  )
+
+  object Activity {
+    implicit val activityDecoder: Decoder[Activity] = deriveConfiguredDecoder
+  }
+
+  sealed trait ActivityType extends Product with Serializable
+
+  object ActivityType {
+    case object Game      extends ActivityType
+    case object Streaming extends ActivityType
+    case object Listening extends ActivityType
+    case object Custom    extends ActivityType
+
+    implicit val activityTypeDecoder: Decoder[ActivityType] = Decoder[Int].emap {
+      case 0     => Right(Game)
+      case 1     => Right(Streaming)
+      case 2     => Right(Listening)
+      case 4     => Right(Custom)
+      case other => Left(s"Unknown activity type ID: $other")
+    }
+  }
+
+  case class Timestamps(start: Timestamp, end: Timestamp)
```

Comment:
I don't love the name Timestamps tbh. What if we did like TimestampRange or something

Reply:
I'm just basing everything off of what the Discord docs call it. It refers to this as "timestamps object" [here](https://discordapp.com/developers/docs/topics/gateway#activity-object). I do agree that your name makes more sense though. So if we decided to name them what _we_  want there's probably a bunch of others that could have better names too.

Reply:
Ok that's fine with me. I think we probably ought to keep it what discord uses so that it is more transparent for our users